snekmer/scripts/learn_learn.py

```diff
@@ -0,0 +1,235 @@
+# ---------------------------------------------------------
+# Imports
+# ---------------------------------------------------------
+
+import pickle
+from datetime import datetime
+from os import makedirs
+from os.path import exists, join
+
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+import snekmer as skm
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import train_test_split, StratifiedKFold
+from sklearn.preprocessing import LabelEncoder
+
+import numpy as np
+import pandas as pd
+import pyarrow as pa
+import pyarrow.csv as csv
+import seaborn as sns
+import sklearn
+from Bio import SeqIO
+from scipy.interpolate import interp1d
+from scipy.stats import rankdata
+import random
+import snekmer as skm
+from scipy.ndimage import gaussian_filter1d
+import sklearn.metrics.pairwise
+import itertools
+import os
+import shutil
+import sys
+import re
+# ---------------------------------------------------------
+# Files and Parameters
+# ---------------------------------------------------------
+
+config = snakemake.config
+
+# change matplotlib backend to non-interactive
+plt.switch_backend("Agg")
+
+outDir = skm.io.define_output_dir(
+    config["alphabet"], config["k"], nested=config["nested_output"]
+)
+
+# ---------------------------------------------------------
+# Run script
+# ---------------------------------------------------------
+
+class Library:
+    """
+Initializes a Library object.
+
+This object is responsible for processing annotations and kmer data.
+It creates a kmer-count matrix for each input file.
+
+Attributes:
+annotation (list): A list to store annotations loaded from files.
```

Comment:
See previous comment about type hinting (since the types are annotated in the docstrings anyway)